cli/replicaset/cmd/roles.go

```diff
@@ -0,0 +1,121 @@
+package replicasetcmd
+
+import (
+	"fmt"
+
+	"github.com/apex/log"
+	"github.com/tarantool/tt/cli/connector"
+	"github.com/tarantool/tt/cli/replicaset"
+	"github.com/tarantool/tt/cli/running"
+	libcluster "github.com/tarantool/tt/lib/cluster"
+)
+
+// RolesAddCtx describes the context to add role to
+// provided config scope.
+type RolesAddCtx struct {
+	// InstName is an instance name in which add or remove role.
+	InstName string
+	// GroupName is an replicaset name in which add or remove role.
+	GroupName string
+	// ReplicasetName is an replicaset name in which add or remove role.
+	ReplicasetName string
+	// IsGlobal is an boolean value if role needs to add in global scope.
+	IsGlobal bool
+	// RoleName is a name of role to add.
+	RoleName string
+	// Publishers is data publisher factory.
+	Publishers libcluster.DataPublisherFactory
+	// Collectors is data collector factory.
+	Collectors libcluster.DataCollectorFactory
+	// IsApplication true if an application passed.
+	IsApplication bool
+	// Orchestrator is a forced orchestator choice.
+	Orchestrator replicaset.Orchestrator
+	// Conn is an active connection to a passed instance.
+	Conn connector.Connector
+	// RunningCtx is an application running context.
+	RunningCtx running.RunningCtx
+	// Force true if unfound instances can be skipped.
+	Force bool
+	// Timeout describes a timeout in seconds.
+	// We keep int as it can be passed to the target instance.
+	Timeout int
+}
+
+// RolesAdd adds role with provided path target to config.
+func RolesAdd(ctx RolesAddCtx) error {
+	var (
+		orchestratorType replicaset.Orchestrator
+		err              error
+	)
+	orchestratorType, err = getInstanceOrchestrator(ctx.Orchestrator, ctx.Conn)
```

Comment:
Why is it done this way?
```suggestion
	orchestratorType, err := getInstanceOrchestrator(ctx.Orchestrator, ctx.Conn)
```